### PR TITLE
Refactor to initialize application context and fix lazy loading of merge nodes

### PR DESCRIPTION
**Pull request by naiquevin**
NOTE: Opening a PR for early feedback. Planning to push more commits. Please don't merge yet. 

- Refactor to add node registry and buffer managers to actix app state
- Lazy-load vectors from disk if required at the time of indexing.

### CODE REVIEW

File: src/models/lazy_load.rs [hunk 1]
```diff
@@ -365,6 +365,9 @@ impl<T: Clone + 'static> LazyItem<T> {
 }
 
 impl<T: Clone + CustomSerialize + 'static> LazyItem<T> {
+    // @TODO: This function uses the `load_item` method of
```
Comment (naiquevin) [inline src/models/lazy_load.rs:368]: What do you think about this?

Comment (a-rustacean) [reply src/models/lazy_load.rs:368]: Good catch - we should be using the `get_object` method here for caching purposes. However, there's another issue in this code that I just realized. By using the `load_item` method, we're directly deserializing `T`. What we should be doing instead is first deserialize the offsets of `T` and `versions`, and then deserialize both `T` and `versions` from their respective offsets. The `get_object` method uses the [lazy_item_deserialize_impl](https://github.com/cosdata/cosdata/blob/main/src/models/serializer/lazy_item.rs#L203) function which handles this correctly. Please update the code accordingly.

Comment (naiquevin) [reply src/models/lazy_load.rs]: I've pushed a commit - https://github.com/cosdata/cosdata/pull/132/commits/647e13759c5e854aae8987975b1ec00062303a6d. Please check.

Comment (naiquevin) [reply src/models/lazy_load.rs]: Please hold on. A fix is needed post rebase.

Comment (naiquevin) [reply src/models/lazy_load.rs]: Fixed.

**Pull request merged**